src/components/Identity/util.js

```diff
@@ -43,4 +43,37 @@ const normalizeCustomerIds = customerIds => {
   }, {});
 };
 
-export { validateCustomerIds, normalizeCustomerIds };
+const bufferToHex = buffer => {
+  return Array.prototype.map
+    .call(new Uint8Array(buffer), item => `00${item.toString(16)}`.slice(-2))
+    .join("");
+};
+
+const encodeText = str => {
+  if (window.TextEncoder) {
+    return new TextEncoder("utf-8").encode(str);
+  }
+  // IE 11, which doesn't have TextEncoder
+  const cleanString = unescape(encodeURIComponent(str));
+  return new Uint8Array(cleanString.split("").map(char => char.charCodeAt(0)));
+};
+
+const sha256Buffer = message => {
+  const data = encodeText(message);
+  const crypto = window.msCrypto || window.crypto;
+  const result = crypto.subtle.digest("SHA-256", data);
+  if (result.then) {
+    return result;
+  }
+  // IE 11, whose result is a CryptoOperation object instead of a promise
+  return new Promise((resolve, reject) => {
+    result.addEventListener("complete", () => {
+      resolve(result.result);
+    });
+    result.addEventListener("error", () => {
+      reject();
+    });
+  });
+};
```

Comment:
I feel like `sha256Buffer` and `bufferToHex` could eventually be useful to other components which, in my mind, qualifies it to be placed up in `core/utils`. I think other people might disagree though, so I've started a thread in the alloy-engineering Slack channel.